modules/core/04-channel/keeper/events.go

```diff
@@ -337,6 +337,28 @@ func emitChannelUpgradeAckEvent(ctx sdk.Context, portID string, channelID string
 	})
 }
 
+// emitChannelUpgradeConfirmEvent emits a channel upgrade confirm event
+func emitChannelUpgradeConfirmEvent(ctx sdk.Context, portID, channelID string, currentChannel types.Channel) {
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeChannelUpgradeConfirm,
+			sdk.NewAttribute(types.AttributeKeyPortID, portID),
+			sdk.NewAttribute(types.AttributeKeyChannelID, channelID),
+			sdk.NewAttribute(types.AttributeKeyChannelState, currentChannel.State.String()),
```

Comment:
We emit the channel state here because the relayer will need it to execute `ChanUpgradeOpen` right?

Reply:
that was idea, though its becoming increasingly unclear to me if relayers are actually using events (I remember rly didnt?)